Sloth.Core/Domain/JobRecord.cs

```diff
@@ -34,6 +34,11 @@ public JobRecord()
 
         public void SetDetails(object details)
         {
+            if (details is string stringDetails)
+            {
+                Details = stringDetails;
+                return;
```

Comment:
I'd prefer just an if/else here, the early return makes sense but isn't super easy to understand when skimming